microbench/src/main/java/alluxio/BaseFileStructure.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import site.ycsb.generator.NumberGenerator;
+import site.ycsb.generator.UniformLongGenerator;
+import site.ycsb.generator.ZipfianGenerator;
+
+import java.util.ArrayList;
+
+/**
+ * Governs the file structure parameters inside a JMH micro bench.
+ */
+@State(Scope.Benchmark)
+public class BaseFileStructure {
+
+  @Param({"0", "1", "10"})
+  public int mDepth;
+
+  @Param({"10", "100", "1000"})
+  public int mFileCount;
+
+  @Param({"UNIFORM", "ZIPF"})
+  public Distribution mDistribution;
+
+  // each depth level needs its own file id generator
+  public ArrayList<NumberGenerator> mFileGenerators = new ArrayList<>();
```

Comment:
I think this is ok, but maybe just incase this variable should be set in the init() function.